### PR TITLE
Improving clusterloader's Measurement API

### PR DESCRIPTION
**Pull request by shyamjvs**
Based on my thoughts from https://github.com/kubernetes/perf-tests/pull/177#issuecomment-409530554

/cc @wojtek-t @krzysied

### CODE REVIEW

File: clusterloader/docs/design.md [hunk 1]
```diff
@@ -278,70 +276,78 @@ be modifying that part of Cluster Loader codebase. Within the codebase, we will
...
 
 At the high level, to implement gathering a given portion of data or measure a new
-SLO, you will need to implement a very simple Go interface, similar to:
+SLO, you will need to implement a very simple Go interface:
 
 ```
 type Measurement interface {
-	StartMeasurement(config MeasurementConfig)
-	StopMeasurement()
+	Execute(config MeasurementConfig) error
```
Comment (krzysied) [inline clusterloader/docs/design.md]: `Execute(config *MeasurementConfig) error`?
I know it's nitpicking, but i want to use this interface (as it is) in code.

**Comment by wojtek-t** [comment]
I like it - thanks!

/lgtm
/approve

File: clusterloader/docs/design.md [hunk 2]
```diff
@@ -278,70 +276,78 @@ be modifying that part of Cluster Loader codebase. Within the codebase, we will
...
-as `measurements` in your test. To be more specific, at the config level
-`Measurement` will be defined as:
+and registering this type in some factory, to enable use of `ResourceGatherer`
+as a measurement 'Method' in your test. And finally, at the config level,
+each `Measurement` is defined as:
 
 ```
 struct Measurement {
-	// A measurement Start/Stop method to be run.
+	// A measurement method to be .
```
Comment (krzysied) [inline clusterloader/docs/design.md]: "run" shouldn't be removed.

**Comment by shyamjvs** [comment]
Fixed above in https://github.com/kubernetes/perf-tests/pull/179.

**Pull request merged**